### PR TITLE
Create eslint.yml

### LINKED ISSUE #10: Create GitHub Action for running a basic ESLint check

### PR DESCRIPTION
**Pull request by nathansit**
Activitate scanning on commit Closes #10

### CODE REVIEW

File: .github/workflows/eslint.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# ESLint is a tool for identifying and reporting on patterns
+# found in ECMAScript/JavaScript code.
+# More details at https://github.com/eslint/eslint
+# and https://eslint.org
+
+name: ESLint
+
+on:
+  push:
+    branches: [ "master" ]
```
Comment (gideontong) [inline .github/workflows/eslint.yml]: All branches please

File: .github/workflows/eslint.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "master" ]
```
Comment (gideontong) [inline .github/workflows/eslint.yml]: All branches please

File: .github/workflows/eslint.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  schedule:
+    - cron: '15 1 * * 3'
```
Comment (gideontong) [inline .github/workflows/eslint.yml]: No need for schedule

File: .github/workflows/eslint.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+jobs:
+  eslint:
+    name: Run eslint scanning
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      security-events: write
+      actions: read # only required for a private repository by github/codeql-action/upload-sarif to get the Action run status
```
Comment (gideontong) [inline .github/workflows/eslint.yml]: Doesn't need this many protections

File: .github/workflows/eslint.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        uses: actions/checkout@v4
+
+      - name: Install ESLint
+        run: |
+          npm install eslint@8.10.0
+          npm install @microsoft/eslint-formatter-sarif@2.1.7
+
+      - name: Run ESLint
+        run: npx eslint .
+          --config .eslintrc.js
```
Comment (gideontong) [inline .github/workflows/eslint.yml]: Wrong config file location

**Review by gideontong — CHANGES_REQUESTED** [review]

File: .github/workflows/eslint.yml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        uses: actions/checkout@v4
+
+      - name: Install ESLint
+        run: |
+          npm install eslint@8.10.0
+          npm install @microsoft/eslint-formatter-sarif@2.1.7
+
+      - name: Run ESLint
+        run: npx eslint .
+          --config .eslintrc.congif.mjs
```
Comment (gideontong) [inline .github/workflows/eslint.yml]: Filename?

**Review by gideontong — CHANGES_REQUESTED** [review]

File: .github/workflows/eslint.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Run ESLint
+        run: npx eslint .
+          --format @microsoft/eslint-formatter-sarif
+          --output-file eslint-results.sarif
+        continue-on-error: true
+
+      - name: Upload analysis results to GitHub
+        uses: github/codeql-action/upload-sarif@v2
+        with:
+          sarif_file: eslint-results.sarif
```
Comment (gideontong) [inline .github/workflows/eslint.yml:39]: This is going to fail because you don't have any permissions

**Review by gideontong — CHANGES_REQUESTED** [review]

File: .github/workflows/eslint.yml (hunk 7 above)
Comment (nathansit) [reply .github/workflows/eslint.yml:39]: Permissions provided

**Review by gideontong — APPROVED** [review]

**Pull request merged**